lib/SILOptimizer/SILCombiner/SILCombinerApplyVisitors.cpp

```diff
@@ -914,7 +924,29 @@ SILInstruction *SILCombiner::createApplyWithConcreteType(
         });
   }
 
-  if (!UpdatedArgs) {
+  bool canUpdateArgs, madeUpdate;
```

Comment:
This whole section of new code was confusing at first. I *think* what it's doing is determining both whether it is possible to rewrite the call at all and whether at least one argument has changed. If not, avoid creating a new call and delete any temporary instructions to avoid the top-level SILCombine driver from infinitely attempting to optimize. If that's right, then the code looks good but please explain that in a comment before new code.

I'm not sure why it's a written as a closure. The results need to be exposed anyway and there's not much else to hide here.

Reply:
I'll factor it out and add a comment. Thanks.